addons/account/models/account_bank_statement.py

```diff
@@ -170,303 +45,191 @@ def _get_move_line_count(self):
-    # -------------------------------------------------------------------------
-
-    @api.constrains('journal_id')
-    def _check_journal(self):
-        for statement in self:
-            if any(st_line.journal_id != statement.journal_id for st_line in statement.line_ids):
-                raise ValidationError(_('The journal of a bank statement line must always be the same as the bank statement one.'))
-
-    def _constrains_date_sequence(self):
-        # Multiple import methods set the name to things that are not sequences:
-        # i.e. Statement from {date1} to {date2}
-        # It makes this constraint not applicable, and it is less needed on bank statements as it
-        # is only an indication and not some thing legal.
-        return
-
-    # -------------------------------------------------------------------------
-    # BUSINESS METHODS
-    # -------------------------------------------------------------------------
-
-    def open_cashbox_id(self):
-        self.ensure_one()
-        context = dict(self.env.context or {})
-        if context.get('balance'):
-            context['statement_id'] = self.id
-            if context['balance'] == 'start':
-                cashbox_id = self.cashbox_start_id.id
-            elif context['balance'] == 'close':
-                cashbox_id = self.cashbox_end_id.id
-            else:
-                cashbox_id = False
-
-            action = {
-                'name': _('Cash Control'),
-                'view_mode': 'form',
-                'res_model': 'account.bank.statement.cashbox',
-                'view_id': self.env.ref('account.view_account_bnk_stmt_cashbox_footer').id,
-                'type': 'ir.actions.act_window',
-                'res_id': cashbox_id,
-                'context': context,
-                'target': 'new'
-            }
-
-            return action
-
-    def button_post(self):
-        ''' Move the bank statements from 'draft' to 'posted'. '''
-        if any(statement.state != 'open' for statement in self):
-            raise UserError(_("Only new statements can be posted."))
-
-        self._check_cash_balance_end_real_same_as_computed()
-
-        for statement in self:
-            if not statement.name:
-                statement._set_next_sequence()
-
-        self.write({'state': 'posted'})
+        Currently it removes the statements with no lines, can be extended if needed
+        """
+        self.search([
+            ('line_ids', '=', False),
+            ('write_date', '<', fields.Date.context_today(self))
+        ]).unlink()
+
+    def _post(self):
+        # if any(stmt.state != 'complete' for stmt in self):
+        #     raise UserError(_("You can only validate statements in the complete state."))
+        self.state = 'posted'
         lines_of_moves_to_post = self.line_ids.filtered(lambda line: line.move_id.state != 'posted')
         if lines_of_moves_to_post:
             lines_of_moves_to_post.move_id._post(soft=False)
```

Comment:
The idea was to have a simpler workflow... This must not be there in my opinion.
Also, `self.line_ids.filtered(lambda line: line.move_id.state == 'draft')`
Also, it must work in multiple recordset to avoid performance issues

Reply:
My idea is to prevent the users from resetting the moves related to st_line to draft. But it has not been discussed yet